content/docs/rfcs/32/README.md

```diff
@@ -0,0 +1,565 @@
+| **Identity secret hash**  | The hash of the identity secret, obtained using the Poseidon hash function. It is used for deriving the identity commitment of the user, and as a private input for zk proof generation. The secret hash should be treated as "private key" for the RLN protocol. The anonymity and security of the user is dependent on the secret hash.                                       |
+| **Identity commitment**   | Hash obtained from the `Identity secret hash` by using the poseidon hash function. It is used by the users for registering in the protocol.                                       |
+| **Signal**                | The message that needs to be sent by the user in a privacy preserving way. The message can be of different type, depending on the application. It can be a chat message, url request, protobuf message. |
+| **Signal hash**           | Hash of the signal, used as an input in the RLN circuit. |
+| **Epoch**                 | An identifier that groups signals and determines the validity of the signal. Can be though ot as voting booth. Usually a timestamp or time interval in the RLN apps. |
+| **RLN Identifier**        | Random finite field value unique per RLN app. It is used for additional cross-application security. The role of the RLN identifier is protection of the user secrets being compromised if signals are being generated with the same credentials at different apps. |
+| **RLN membership tree**   | Merkle tree data structure, filled with identity commitments of the users. Serves as a data structure that ensures user registrations.   |
+| **Merkle proof**          | Proof that a user is member of the RLN membership tree.    |
+
+
+## RLN ZK-Circuit specific terms
+
+| Term           | Description                                                       |
+|---------------------------|-------------------------------------------------------------------------------------|
+| **X Share**               | Hash of the signal, public input in the RLN circuit.    |
+| **A0**               | The identity secret hash.    |
+| **A1**               | Poseidon hash of [a0, external_nullifier] (see about External nullifier below).    |
+| **Y Share**          | The result of the polynomial equation (y = a0 + a1*x). Public output of the circuit.    |
+| **External nullifier**    | `keccak256` hash of the Epoch.    |
+| **Internal nullifier**    | Poseidon hash of [a1, rln_identifier]. This field ensures that a user can send only one valid signal per epoch without risking being slashed. Public output of the circuit.   |
+
+
+
+## ZK Circuits specification
+
+Anonymous signaling with spam detection is enabled by proving that the user is part of a group which has high barriers to entry (form of stake) and 
+enabling secret reveal if more signals are produced than the spam threshold (system parameter) per epoch.
```

Comment:
I'd suggest focusing on the general application of RLN, and replacing the term `spam`  with `double signaling`, also using `rate limit` or `double signaling detection` instead of `spam detection`.

Reply:
I agree with your suggestion, it's much more precise.